command/agent/operator_endpoint_test.go

```diff
@@ -56,3 +57,202 @@ func TestOperator_OperatorRaftPeer(t *testing.T) {
 		}
 	})
 }
+
+func TestOperator_KeyringInstall(t *testing.T) {
+	oldKey := "H3/9gBxcKKRf45CaI2DlRg=="
+	newKey := "z90lFx3sZZLtTOkutXcwYg=="
+	configFunc := func(c *Config) {
+		c.EncryptKey = oldKey
+	}
+	httpTestWithConfig(t, func(srv *HTTPServer) {
+		body := bytes.NewBufferString(fmt.Sprintf("{\"Key\":\"%s\"}", newKey))
+		req, err := http.NewRequest("POST", "/v1/operator/keyring", body)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+
+		resp := httptest.NewRecorder()
+		_, err = srv.OperatorKeyringEndpoint(resp, req)
+		if err != nil {
+			t.Fatalf("err: %s", err)
+		}
+
+		listResponse, err := srv.agent.ListKeys("")
+		if err != nil {
+			t.Fatalf("err: %s", err)
+		}
+		if len(listResponse.Responses) != 2 {
+			t.Fatalf("bad: %d", len(listResponse.Responses))
+		}
+
+		for _, response := range listResponse.Responses {
+			count, ok := response.Keys[newKey]
+			if !ok {
+				t.Fatalf("bad: %v", response.Keys)
+			}
+			if count != response.NumNodes {
+				t.Fatalf("bad: %d, %d", count, response.NumNodes)
+			}
+		}
+	}, configFunc)
+}
+
+func TestOperator_KeyringList(t *testing.T) {
+	key := "H3/9gBxcKKRf45CaI2DlRg=="
+	configFunc := func(c *Config) {
+		c.EncryptKey = key
+	}
+	httpTestWithConfig(t, func(srv *HTTPServer) {
+		req, err := http.NewRequest("GET", "/v1/operator/keyring", nil)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+
+		resp := httptest.NewRecorder()
+		r, err := srv.OperatorKeyringEndpoint(resp, req)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+		responses, ok := r.([]*structs.KeyringResponse)
+		if !ok {
+			t.Fatalf("err: %v", !ok)
+		}
+
+		// Check that we get both a LAN and WAN response, and that they both only
+		// contain the original key
+		if len(responses) != 2 {
+			t.Fatalf("bad: %d", len(responses))
+		}
+
+		// WAN
```

Comment:
Yes, it's consistent; [here](https://github.com/hashicorp/consul/blob/master/consul/internal_endpoint.go#L124-L132) we call executeKeyringOp with `wan = true`, then with `wan = false`, and in this test we always get back the wan reply first.